etl/test/test_etl_load.py

```diff
@@ -1,94 +1,117 @@
+            'collection_specimen': 'specimen',
         },
-        "participant_meta_map": {},
-        "assay_meta_map": {},
-        "qc_meta_map": {}
+        'participant_meta_map': {},
+        'assay_meta_map': {},
+        'qc_meta_map': {},
+    },
+    'data': {
+        'sample_id': '123456',
+        'external_id': 'AAA000000',
+        'individual_id': '678',
+        'sequencing_type': 'exome',
+        'collection_centre': 'ABCDEF',
+        'collection_date': '2023-08-05T01:39:28.611476',
+        'collection_specimen': 'blood',
     },
-    "data":
-    {
-        "sample_id": "123456",
-        "external_id": "AAA000000",
-        "individual_id": "678",
-        "sequencing_type": "exome",
-        "collection_centre": "ABCDEF",
-        "collection_date": "2023-08-05T01:39:28.611476",
-        "collection_specimen": "blood"
-    }
 }
-"""
 
 
-class TestEtlLoad(DbIsolatedTest):
+class TestGetParserInstance(GenericMetadataParser):
+    def __init__(
+        self,
+        project: str,
+    ):
+        super().__init__(
+            project=project,
+            sample_name_column='sample',
+            search_locations=[],
+            participant_meta_map={},
+            sample_meta_map={},
+            assay_meta_map={},
+            qc_meta_map={},
+            allow_extra_files_in_search_path=False,
+        )
+
+    def get_sample_id(self, row) -> str:
+        """Get external sample ID from row"""
+        return row['sample']
+
+    @staticmethod
+    def get_info() -> tuple[str, str]:
+        """
+        Information about parser, including short name and version
+        """
+        return ('test', 'v1')
+
+
+class TestEtlLoad(TestCase):
     """Test etl cloud functions"""
 
     @run_as_sync
     async def setUp(self) -> None:
         super().setUp()
 
+    @staticmethod
+    def get_query_job_result(
+        mock_bq,
+        request_type: str | None = None,
+        body: str | None = None,
+        submitting_user: str | None = None,
+    ):
+
+        query_job_result = MagicMock(args={}, spec=['__iter__', '__next__'])
+
+        if request_type or body or submitting_user:
+            query_row = MagicMock(args={}, spec=['body', 'type', 'submitting_user'])
+            query_row.body = body
+            query_row.type = request_type
+            query_row.submitting_user = submitting_user
+
+            query_job_result.total_rows = 1
+            query_job_result.__iter__.return_value = [query_row]
+        else:
+            query_job_result.total_rows = 0
+            query_job_result.__iter__.return_value = []
+
+        mock_bq.return_value.query.return_value.result.return_value = query_job_result
+
     @run_as_sync
-    @patch('etl.load.main.bq.Client', autospec=True)
-    async def test_etl_load_not_found_record(self, bq_client):
-        """Test etl load not found"""
+    @patch('etl.load.main._get_bq_client', autospec=True)
+    async def test_etl_load_not_found_record(self, mock_bq):
```

Comment:
Maybe this test does not need to be async as the rest of them, just to be consistent?

Reply:
Totally agree :)